File: ReadFile/Read.py
"""
    Author : Pakshal Shashikant Jain 
    Date : 19/05/2024
    Program : Write application which accept file name from user and read all data from that file
              and display contents on screen.
            
              Input : Demo.txt
              Output : Display all data of file. 
"""

#############################################################################################################################################################

#Function To Read Whole Data of File
def ReadFile(fname) :
    fd = 0;
    sret = " ";
    isize = 0;

    #Opening File in Read Mode
    fd = open(fname,mode = 'r',encoding = 'UTF-8');

    #Logic To Read File
    if(fd == None) :
        print("Unable To Open File");
    else :
        print("File Successfully Opened with File Descriptor :",fd.fileno());

    #Reading Data of File 
    sret = fd.read();
    print("Data From File is : ");
    print(sret);

    #Calculating Size of File 
    isize = fd.seek(0,2);

    #Closing Opened File
    fd.close();

    return isize;

############################################################################################################################

#Main Function
def main() :
    name = " ";
    iret = 0;

    print("Jay Ganesh....");

    #Taking File Name as Input From User
    print("Enter File Name : ");
    name = input();

    #Call To ReadFile Function
    iret = ReadFile(name);
    print("Size of File is : %d bytes"%iret);

################################################################################################################################################

#Call Back Method To Call Main Function
if __name__ == "__main__" :
    #Call To Main Function
    main();

File: ReadFile/Read.c
/*
    Author : Pakshal Shashikant Jain 
    Date : 19/05/2024
    Program : Write application which accept file name from user and read all data from that file
              and display contents on screen.
            
              Input : Demo.txt
              Output : Display all data of file. 
*/

////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Required Libraries
#include<stdio.h>
#include<fcntl.h>
#include<stdlib.h>
#include<unistd.h>

////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Function To Read File
int ReadFile(char fname[])
{
    int fd = open(fname,O_RDONLY | O_CREAT);
    int ret = 0;
    char arr[40] = {'\0'};
    int i = 0;
    
    //Logic To Read Whole File
    if(fd == -1)
    {
        printf("Unable To Create file Or to Open File");
    }
    else 
    {
        printf("File Successfully Created or Opened With File Descriptor : %d\n",fd);
    }
    
    printf("Data From File is : \n");
    //Reading Whole File
    while((ret = read(fd,arr,sizeof(fd))) != 0)
    {
        write(1,arr,ret);  
    }

    //Getting Size of File
    ret = lseek(fd,0,SEEK_END);
    
    //Closing Opened File
    close(fd);

    //Returing Size of File
    return ret;
}

///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Main Function
int main()
{
    char name[40] = {'\0'};
    int ret = 0;

    printf("Jay Ganesh....\n");

    //Taking File Name as Input From User
    printf("Enter File Name : \n");
    scanf("%[^'\n']s",&name);

    //Call To Read File Function
    ret = ReadFile(name);

    //Printing Size of File
    printf("\nSize of File is : %d bytes",ret);

    return 0;
}